torch/quantization/fx/utils.py

```diff
@@ -360,7 +360,8 @@ def create_getattr_from_value(module: torch.nn.Module, graph: Graph, prefix: str
     """
     get_new_attr_name = get_new_attr_name_with_prefix(prefix)
     attr_name = get_new_attr_name(module)
-    module.register_buffer(attr_name, torch.tensor(value))
+    device = assert_and_get_unique_device(module)
```

Comment:
Is this solution specific to fx? If we want this approach to work for eager, would we need to modify the from_float method?

Reply:
All of the changes I made affect the prepare code at the point where it has called the non-fx prepare code so it should be applicable to both. There may be a slight differece but it should be easy to make work on eager, at least I cannot see a reason why it wouldn't.